iina-plugin/main.swift

```diff
@@ -0,0 +1,430 @@
+}
+
+// MARK: - Utilities
+
+fileprivate var appSupportDir: URL {
+  let appSupportPath = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask)
+  let bundleID = Bundle.main.bundleIdentifier ?? "com.colliderli.iina"
+  return appSupportPath.first!.appendingPathComponent(bundleID).appendingPathComponent("plugins")
+}
+
+fileprivate func url(_ file: String, in dir: URL) -> URL {
+  return dir.appendingPathComponent(file).standardized
+}
+
+fileprivate func createParentDirectory(_ path: URL) throws {
+  let parent = path.deletingLastPathComponent()
+  if FileManager.default.fileExists(atPath: parent.path) {
+    return
+  }
+  try FileManager.default.createDirectory(at: parent, withIntermediateDirectories: true)
+}
+
+fileprivate func linesByRemovingEmptyLines(_ document: String) -> [String] {
+  return document.split(separator: "\n")
+    .map { $0.replacingOccurrences(of: "\\s+$", with: "", options: .regularExpression) }
+    .filter { !$0.isEmpty }
+}
+
+fileprivate func resolvePluginDir(from path: String) -> URL? {
+  let pluginDir: URL?
+  if path.hasPrefix("/") {
+    pluginDir = URL(string: path)?.standardized
+  } else {
+    pluginDir = URL(string: path, relativeTo: currentDirURL)?.standardized
+  }
+  return pluginDir
+}
+
+fileprivate extension Process {
+  @discardableResult
+  static func run(_ cmd: [String], at currentDir: URL? = nil) -> (process: Process, stdout: Pipe, stderr: Pipe) {
+    guard cmd.count > 0 else {
+      fatalError("Process.launch: the command should not be empty")
+    }
+
+    let (stdout, stderr) = (Pipe(), Pipe())
+    let process = Process()
+    if #available(macOS 10.13, *) {
+      process.executableURL = URL(fileURLWithPath: cmd[0])
+      process.currentDirectoryURL = currentDir
+    } else {
+      process.launchPath = cmd[0]
+      if let path = currentDir?.path {
+        process.currentDirectoryPath = path
+      }
+    }
+    process.arguments = [String](cmd.dropFirst())
+    process.standardOutput = stdout
+    process.standardError = stderr
+    process.launch()
+    process.waitUntilExit()
+
+    return (process, stdout, stderr)
+  }
+}
+
+// MARK: - Command line helpers
+
+func promptYesOrNo(_ message: String, defaultValue: Bool = false) -> Bool {
+  print("\(message) (y/n, default: \(defaultValue ? "y" : "n")) ", terminator: "")
+  guard let answer = readLine() else { return defaultValue }
+  let trimmedAnswer = answer.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
+  if ["y", "yes"].contains(trimmedAnswer) {
+    return true
+  }
+  if ["n", "no"].contains(trimmedAnswer) {
+    return false
+  }
+  return defaultValue
+}
+
+
+func prompt(_ message: String, chooseFrom choices: [String]) -> String {
+  assert(!choices.isEmpty)
+  print(message)
+  while true {
+    for (i, choice) in choices.enumerated() {
+      print("  \(i + 1): \(choice)")
+    }
+    print("Input a number (default: 1): ", terminator: "")
+    if let answer = readLine() {
```

Comment:
if answer.isEmpty {
        return choices[0]
      }

Reply:
Because that check is missing defaulting is not working:
```text
Use a frontend framework to build the user interface?
  1: None
  2: React
  3: Vue
Input a number (default: 1): 
  1: None
  2: React
  3: Vue
Input a number (default: 1): 
```